File: src/main/java/com/analysisgg/modules/riotapi/adapter/in/web/PlayerAnalyticsResponse.java
package com.analysisgg.modules.riotapi.adapter.in.web;

import java.util.List;

public record PlayerAnalyticsResponse(
        String puuid,
        String gameName,
        String tagLine,
        String region,
        int profileIconId,
        long summonerLevel,
        RankedQueuesResponse rankedQueues,
        List<MatchResponse> matches,
        List<PastSeasonRankResponse> pastSeasonRanks
) {
    public PlayerAnalyticsResponse(
            String puuid,
            String gameName,
            String tagLine,
            String region,
            RankedQueuesResponse rankedQueues,
            List<MatchResponse> matches
    ) {
        this(puuid, gameName, tagLine, region, 29, 0L, rankedQueues, matches, List.of());
    }

    public PlayerAnalyticsResponse(
            String puuid,
            String gameName,
            String tagLine,
            String region,
            List<MatchResponse> matches
    ) {
        this(puuid, gameName, tagLine, region, null, matches);
    }

    public PlayerAnalyticsResponse {
        if (rankedQueues == null) {
            rankedQueues = new RankedQueuesResponse(
                    new RankedQueueSummaryResponse("RANKED_SOLO_5x5", null, null, null, null, null, null),
                    new RankedQueueSummaryResponse("RANKED_FLEX_SR", null, null, null, null, null, null)
            );
        }
        if (matches != null) {
            matches = List.copyOf(matches);
        } else {
            matches = List.of();
        }
        if (pastSeasonRanks != null) {
            pastSeasonRanks = List.copyOf(pastSeasonRanks);
        } else {
            pastSeasonRanks = List.of();
        }
    }
}

File: src/main/frontend/src/features/dashboard/infrastructure/api/PlayerAnalyticsResponse.ts
import type { MatchSummary } from '../../domain/MatchSummary';

export interface RankedQueueSummary {
  queueType: string;
  tier: string | null;
  rank: string | null;
  leaguePoints: number | null;
  wins: number | null;
  losses: number | null;
  winRate: number | null;
}

export interface RankedQueues {
  soloDuo: RankedQueueSummary;
  flex: RankedQueueSummary;
}

export interface PastSeasonRank {
  season: string;
  tier: string;
  rank: string | null;
}

export interface PlayerAnalyticsResponse {
  puuid: string;
  gameName: string;
  tagLine: string;
  region: string;
  profileIconId: number;
  summonerLevel: number;
  rankedQueues: RankedQueues;
  matches: MatchSummary[];
  pastSeasonRanks: PastSeasonRank[];
}
